### ISSUE TITLE
twistd man page needs a section on signals

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7307]
|<img alt="anthony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| anthony reported|
|-|-|
|Trac ID|trac#689|
|Type|enhancement|
|Created|2004-08-27 14:32:39Z|
|Branch|https://github.com/twisted/twisted/tree/twisted-man-signals-689|

<details><summary>Searchable metadata</summary>

```
trac-id__689 689
type__enhancement enhancement
reporter__anthony anthony
priority__normal normal
milestone__ 
branch__branches_twisted_man_signals_689 branches/twisted-man-signals-689
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1093617159000000 1093617159000000
changetime__1264109075000000 1264109075000000
version__ 
owner__ 
cc__anthony cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="anthony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|anthony commented|
|-|-|

```
#!html
<pre>
doc/man/twistd.1 needs a short section on signals. A running twistd accepts
SIGINT for a clean shutdown, and SIGUSR1 for 'rotate log files'.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [24470]) Branching to 'twisted-man-signals-689'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

There seems to be precedent for a dedicated `SIGNALS` section (eg bash, Xserver, skill, snice, epic4, etc).

I'd put this info into such a section.  Also note there's an existing mention of signals in the docs for the `--debug` option in that man page.  The docs there are incomplete, since both SIGINT and SIGUSR2 drop into the debugger.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_6):
> There seems to be precedent for a dedicated `SIGNALS` section (eg bash, Xserver, skill, snice, epic4, etc).

Haven't checked the source or google yet, but where would i find this info? Is it specific to twistd?

>
> I'd put this info into such a section.  Also note there's an existing mention of signals in the docs for the `--debug` option in that man page.  The docs there are incomplete, since both SIGINT and SIGUSR2 drop into the debugger.

Made the suggested change in r24538

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Replying to [thijs](#note_7):
> Replying to [exarkun](#note_6):
> > There seems to be precedent for a dedicated `SIGNALS` section (eg bash, Xserver, skill, snice, epic4, etc).
>
> Haven't checked the source or google yet, but where would i find this info? Is it specific to twistd?

Something like the list below?

```
$ python signal_getsignal.py
SIGHUP     ( 1): SIG_DFL
SIGINT     ( 2): <built-in function default_int_handler>
SIGQUIT    ( 3): SIG_DFL
SIGILL     ( 4): SIG_DFL
SIGTRAP    ( 5): SIG_DFL
SIGIOT     ( 6): SIG_DFL
SIGEMT     ( 7): SIG_DFL
SIGFPE     ( 8): SIG_DFL
SIGKILL    ( 9): None
SIGBUS     (10): SIG_DFL
SIGSEGV    (11): SIG_DFL
SIGSYS     (12): SIG_DFL
SIGPIPE    (13): SIG_IGN
SIGALRM    (14): <function alarm_received at 0x7c3f0>
SIGTERM    (15): SIG_DFL
SIGURG     (16): SIG_DFL
SIGSTOP    (17): None
SIGTSTP    (18): SIG_DFL
SIGCONT    (19): SIG_DFL
SIGCHLD    (20): SIG_DFL
SIGTTIN    (21): SIG_DFL
SIGTTOU    (22): SIG_DFL
SIGIO      (23): SIG_DFL
SIGXCPU    (24): SIG_DFL
SIGXFSZ    (25): SIG_IGN
SIGVTALRM  (26): SIG_DFL
SIGPROF    (27): SIG_DFL
SIGWINCH   (28): SIG_DFL
SIGINFO    (29): SIG_DFL
SIGUSR1    (30): SIG_DFL
SIGUSR2    (31): SIG_DFL
```

Generated with the script from this [place](http://blog.doughellmann.com/2008/08/pymotw-signal.html):

```python
import signal
import pprint

def alarm_received(n, stack):
    return

signal.signal(signal.SIGALRM, alarm_received)

signals_to_names = {}
for n in dir(signal):
    if n.startswith('SIG') and not n.startswith('SIG_'):
        signals_to_names[getattr(signal, n)] = n

for s in xrange(1, signal.NSIG):
    name = signals_to_names[s]
    handler = signal.getsignal(s)
    if handler is signal.SIG_DFL:
        handler = 'SIG_DFL'
    elif handler is signal.SIG_IGN:
        handler = 'SIG_IGN'
    print '%-10s (%2d):' % (name, s), handler
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Delete the winefish-related additions.  The quotes around "rotate log files" aren't really necessary.  Also, I'd put the signals section directly above the author section.  Options are more interesting than signals.  For some reason, the "SIGNALS" header is rendering indented in my man reader.  Do you see that as well?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

Bump.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_12"></a>|@jesstess commented|
|-|-|

(In [28023]) Move SIGNALS to just above AUTHORS and other cosmetic changes.

refs [#7307](https://github.com/twisted/twisted/issues/7307)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set owner to @jesstess|
|-|-|

Can you link to http://twistedmatrix.com/trac/wiki/TwistedDevelopment#FilingTickets instead of http://twistedmatrix.com/bugs?

Can you also update the copyright in twisted.copyright as it's used in the cmdline tools as well. IMO it doesn't really matter in what kind of ticket that is date is updated, opening a new ticket is overkill..

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_15"></a>|@jesstess commented|
|-|-|

(In [28047]) Change bug link on man page and bump copyrights.

refs [#7307](https://github.com/twisted/twisted/issues/7307)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_16"></a>|@jesstess removed owner|
|-|-|

I don't totally love a link to TwistedDevelopment since a bunch of information assuming that you're really interested in contributing instead of just wanting to submit a bug is thrown at you, but I agree that the original link is less helpful and I don't have suggestions for alternatives. I opted for 
http://twistedmatrix.com/trac/wiki/TwistedDevelopment#DevelopmentProcess so that you can see the comment about needing to register before submitting a ticket and have the link to http://twistedmatrix.com/trac/newticket right in front of you.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_17"></a>|@khorn set owner to @jesstess|
|-|-|

It occurs to me that perhaps that wiki page should have a section at the top called "Reporting Bugs" (or perhaps there is a more appropriate place for it somewhere else) which focuses just on the process of reporting bugs, filing tickets, talks about what makes for a good bug report, etc.

At any rate this type of thing should be recorded somewhere.

But that's probably a ticket for another day.

I agree with your link location for now, it's better than just linking to the page.

I don't see any other problems, I think this should be merged.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_18"></a>|@jesstess set status to `closed`|
|-|-|

(In [28049]) Merge twisted-man-signals-689

Authors: thijs, jesstess
Reviewers: exarkun, khorn
Fixes: [#7307](https://github.com/twisted/twisted/issues/7307)

The twistd man page now has a SIGNALS section.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
